.github/workflows/pre-commit.yml

```diff
@@ -13,3 +13,18 @@ jobs:
     - uses: pre-commit/action@v2.0.0
       env:
         SKIP: rstcheck
+
+    - name: "Generate patch file"
+      if: failure()
+      run: |
+        git diff-index -p HEAD > "${PATCH_FILE}"
+        [ -s "${PATCH_FILE}" ] && echo "UPLOAD_PATCH_FILE=${PATCH_FILE}" >> "${GITHUB_ENV}"
```

Comment:
What does [ -s ... ] do?
I think it is anyways much more straight forward to define common environment variables in the header section if the whole file.

Reply:
`-s` is true if the file exists and is not empty. This ensures that the patch file is only uploaded when necessary.

Reply:
> Oh I have finally found it.
> It was as hard to find because google is not good in finding a single letter expression:
> https://tldp.org/LDP/abs/html/fto.html

Next time, you can type `man test`, then `/-s` 🙂

> I think we can remove this hard to read code anyway by a global environment variables for the patch name in the header section. Even without the patch the upload is skipped if the patch file dies not exist alreadautomatically shipped if the patch file dies not exist. 

This is not equivalent. This check ensures that the file is also not empty (because it's possible that pre-commit fails but is unable to make automated changes).